Model/Service/Index.php

```diff
@@ -264,6 +290,8 @@ public function handleProductSync(NostoIndexCollection $collection, Store $store
                 }
                 ++$currentPage;
             } while ($currentPage <= $pages);
+        } catch (MemoryOutOfBoundsException $e) {
+            throw $e;
```

Comment:
This logic is redundant. If you want this exception to be thrown from this method, simply don't catch it.